### PR TITLE
Add-o pipefail for all bash scripts

### LINKED ISSUE #142380: Add a bash script linter for -o pipefail
The idea comes from https://github.com/pytorch/pytorch/pull/142377, it's useful to have `-o pipefail` prevent a similar failure in other bash scripts.

There is also `-u` that ExecuTorch is using on their repo that we could adopt too.

cc @seemethere @malfet @pytorch/pytorch-dev-infra @ezyang

### PR DESCRIPTION
**Pull request by Grove-YJX**
Fixes #142380
I have added -o pipefail in all bash scripts in pytorch/.ci/pytorch. Sorry I didn't double-check the submodule in my last PR. Thanks for the correction! Please contact me again if there are any problems with this fix^^. (Actually contributing to the open source community is an assignment for one of my courses and today is the deadline so I rushed to revise it when I saw an email early in the morning. Haha.)
 @ezyang @malfet @huydhn @zou3519

### CODE REVIEW

File: .ci/pytorch/test.sh [hunk 1]
```diff
@@ -1563,4 +1563,4 @@ else
   test_custom_backend
   test_torch_function_benchmark
   test_benchmarks
-fi
+fi
```
Comment (ezyang) [inline .ci/pytorch/test.sh:1566]: lost trailing newline

**Review by ezyang — APPROVED** [review]

**Review by huydhn — APPROVED** [review]
LGTM!

**Pull request closed without merging**